dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFunderDataProvider.java

```diff
@@ -0,0 +1,190 @@
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.external.provider.impl;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import javax.xml.bind.JAXBException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.dto.MetadataValueDTO;
+import org.dspace.external.model.ExternalDataObject;
+import org.dspace.external.provider.ExternalDataProvider;
+import org.openaire.funders.jaxb.model.OpenAIREHandler;
+import org.openaire.funders.jaxb.model.Project;
+import org.springframework.beans.factory.annotation.Required;
+import org.openaire.funders.jaxb.model.Response;
+import org.openaire.funders.jaxb.model.Results;
+import org.openaire.funders.jaxb.model.Result;
+
+
+/**
+ * This class is the implementation of the ExternalDataProvider interface that
+ * will deal with the OpenAIRE API Funders Data lookup
+ */
+public class OpenAIREFunderDataProvider implements ExternalDataProvider {
+
+    private static Logger log = LogManager.getLogger(OpenAIREFunderDataProvider.class);
+
+    private String sourceIdentifier;
+    private String openaireAPIUrl;
+
+    @Override
+    public String getSourceIdentifier() {
+        return sourceIdentifier;
+    }
+
+    /**
+     * Initialize the class
+     *
+     * @throws java.io.IOException passed through from HTTPclient.
+     */
+    public void init() throws IOException {
+
+    }
+
+    /**
+     * Makes an instance of the OpenAIREFunderDataProvider class based on the
+     * provided parameters. This constructor is called through the spring bean
+     * initialization
+     */
+    private OpenAIREFunderDataProvider(String url) {
+        this.openaireAPIUrl = url;
+    }
+
+    @Override
+    public Optional<ExternalDataObject> getExternalDataObject(String id) {
+        Response response = null;
+        try {
+            response = getResponse(id);
+        } catch (MalformedURLException ex) {
+            java.util.logging.Logger.getLogger(OpenAIREFunderDataProvider.class.getName()).log(Level.SEVERE, null, ex);
+        } catch (JAXBException ex) {
+            java.util.logging.Logger.getLogger(OpenAIREFunderDataProvider.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        ExternalDataObject externalDataObject = convertToExternalDataObject(response);
+        return Optional.of(externalDataObject);
+    }
+
+    protected ExternalDataObject convertToExternalDataObject(Response response) {
+        ExternalDataObject externalDataObject = new ExternalDataObject(sourceIdentifier);
+        Results results = response.getResults();
+
+        Result result = results.getResult();
+        Project project = response.getResults().getResult().getMetadata().getEntity().getProject();
```

Comment:
With this kind of chained gets, we might requesting data on NULL objects. I think the case of not having data, or parts of it, should be addressed. Perhaps throwing exceptions to deal with this later.